Update torch.distributed.DistBackendError type

H-Huang opened the pull request:
Summary: Update torch.distributed.DistBackendError type based on https://fb.workplace.com/groups/pyreqa/posts/5753993921357059

Test Plan:
Pyre tests should pass?

let sandcastle run

Reviewed By: markkm

Differential Revision: D41384130

awgu commented on test/distributed/test_c10d_nccl.py:
```diff
@@ -1031,9 +1031,12 @@ def test_nccl_dist_backend_error(self):
             dist.broadcast(torch.tensor([1, 2, 3]).cuda(), 0)
 
+        self.assertIsInstance(cm.exception, RuntimeError)
+        self.assertIsInstance(cm.exception, Exception)
```
```
>>> re = RuntimeError()
>>> isinsstance(re, Exception)
True
```

Is the second `assertIsInstance` redundant?

H-Huang replied:
good point

awgu reviewed (approved):
Stamp to unblock. That test plan though 🤔

H-Huang committed:
Update torch.distributed.DistBackendError type (#89235)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/89235

Update torch.distributed.DistBackendError type based on https://fb.workplace.com/groups/pyreqa/posts/5753993921357059

Test Plan:
Pyre tests should pass?

let sandcastle run

Reviewed By: awgu, markkm

Differential Revision: D41384130

fbshipit-source-id: 7d5f2c46b5e72751261804a55fb7d32c94b4c21b
